src/quo2/components/dropdowns/dropdown/view.cljs

```diff
@@ -0,0 +1,79 @@
+(ns quo2.components.dropdowns.dropdown.view
+  (:require [quo2.components.dropdowns.dropdown.properties :as properties]
+            [quo2.components.dropdowns.dropdown.style :as style]
+            [quo2.components.icon :as icon]
+            [quo2.components.markdown.text :as text]
+            [quo2.theme :as theme]
+            [react-native.blur :as blur]
+            [react-native.core :as rn]))
+
+(defn- view-internal
+  [{:keys [type size state background on-press icon-name icon? emoji? accessibility-label]
+    :or   {type      :grey
+           size      :size-40
+           state     :default
+           icon?     false
+           emoji?    false
+           icon-name :i/placeholder}
+    :as   props} children]
+  (let [{:keys [icon-size text-size emoji-size]
+         :as   size-properties} (properties/sizes size)
+        {:keys [left-icon-color right-icon-color label-color blur-type
```

Comment:
I think it's quite confusing to have `left-icon-color` and `right-icon-color` so symmetrical, yet have different types. I only realized this because I was going to make a comment about using assoc instead of merge in the map `right-icon-color`.

- `left-icon-color` is used as a single color, a string/keyword.
- `right-icon-color` is a map of (up to) 2 colors.

When things end with `-color`, we automatically assume they are actual colors (strings or even keywords), but certainly not maps.

I see two good solutions:

1. According to how this PR is using `right-icon-color`, it's in fact allowing callers to overwrite any right icon property (it even uses `merge`), so we could rename to `right-icon-props`.

2. If you want to be extra precise that you only take colors, you could also model the component to match the icon API, thus receive `right-icon-color` (string) and `right-icon-color-2` (string).

Reply:
Thank you @smohamedjavid 🚀